botocore/parsers.py

```diff
@@ -761,6 +767,185 @@ def _parse_body_as_json(self, body_contents):
+        elif major_type == 2:
+            return self._parse_byte_string(stream, additional_info)
+        elif major_type == 3:
+            return self._parse_text_string(stream, additional_info)
+        elif major_type == 4:
+            return self._parse_array(stream, additional_info)
+        elif major_type == 5:
+            return self._parse_map(stream, additional_info)
+        elif major_type == 6:
+            return self._parse_tag(stream, additional_info)
+        elif major_type == 7:
+            return self._parse_simple_and_float(stream, additional_info)
+        else:
+            raise ResponseParserError(
+                f"Unsupported inital byte found for data item- "
+                f"Major type:{major_type}, Additional info: "
+                f"{additional_info}"
+            )
+
+    def _parse_tag(self, stream, additional_info):
+        tag = self._parse_integer(stream, additional_info)
+        value = self.parse_data_item(stream)
+        if tag == 1:  # Epoch-based date/time in milliseconds
+            return self._parse_datetime(value)
+        else:
+            raise ResponseParserError(f"Unknown CBOR tag: {tag}")
+
+    def _parse_datetime(self, value):
+        if isinstance(value, (int, float)):
+            return self._timestamp_parser(value)
+        else:
+            raise ResponseParserError(
+                f"Unable to parse datetime value: {value}"
+            )
+
+    def _parse_integer(self, stream, additional_info):
+        if additional_info < 24:
+            return additional_info
+        elif additional_info == 24:
+            return self._read_bytes_as_int(stream, 1)
+        elif additional_info == 25:
+            return self._read_bytes_as_int(stream, 2)
+        elif additional_info == 26:
+            return self._read_bytes_as_int(stream, 4)
+        elif additional_info == 27:
+            return self._read_bytes_as_int(stream, 8)
+        else:
+            raise ResponseParserError(
+                "Invalid additional information for integer"
+            )
+
+    def _parse_byte_string(self, stream, additional_info):
+        if additional_info == 31:
+            chunks = []
+            while True:
+                chunk = self._read_chunk(stream)
+                if chunk is None:
+                    break
+                chunks.append(chunk)
+            return b''.join(chunks)
+        else:
+            length = self._parse_integer(stream, additional_info)
+            return self._read_from_stream(stream, length)
+
+    def _parse_text_string(self, stream, additional_info):
+        return self._parse_byte_string(stream, additional_info).decode('utf-8')
+
+    def _parse_array(self, stream, additional_info):
+        if additional_info == 31:
+            items = []
+            while True:
+                initial_byte = self._read_bytes_as_int(stream, 1)
+                if initial_byte == 0xFF:
```

Comment:
Maybe we should make `0xFF` into a constant to make it more obvious this is a break/terminal byte